compiler/src/iree/compiler/Dialect/HAL/Target/LLVM/CMakeLists.txt

```diff
@@ -143,19 +143,19 @@ iree_cc_library(
 
 add_library(IREELLVMCPUTargetDeps INTERFACE)
 
-function(_add_optional_llvm_dep dep)
-  if(TARGET ${dep})
-    target_link_libraries(IREELLVMCPUTargetDeps INTERFACE ${dep})
+function(_add_optional_llvm_target_deps target)
+  # Key off of the CodeGen target and then include the rest.
+  if(TARGET "LLVM${target}CodeGen")
+    target_link_libraries(IREELLVMCPUTargetDeps INTERFACE
+      "LLVM${target}AsmParser"
+      "LLVM${target}CodeGen"
+      "LLVM${target}Desc"
+      "LLVM${target}Info")
   endif()
 endfunction()
 
-_add_optional_llvm_dep(LLVMAArch64AsmParser)
-_add_optional_llvm_dep(LLVMAArch64CodeGen)
-_add_optional_llvm_dep(LLVMARMAsmParser)
-_add_optional_llvm_dep(LLVMARMCodeGen)
-_add_optional_llvm_dep(LLVMRISCVAsmParser)
-_add_optional_llvm_dep(LLVMRISCVCodeGen)
-_add_optional_llvm_dep(LLVMWebAssemblyAsmParser)
-_add_optional_llvm_dep(LLVMWebAssemblyCodeGen)
-_add_optional_llvm_dep(LLVMX86AsmParser)
-_add_optional_llvm_dep(LLVMX86CodeGen)
+_add_optional_llvm_target_deps(AArch64)
```

Comment:
Could we use `LLVM_TARGETS_TO_BUILD` here instead? Then we don't have to check on whether the target exists and we don't have to hardcode the targets here. This is also missing some of the targets we conditionally enable: `AMDGPU` and `NVPTX`. Is this intentionally only CPU targets?

Reply:
I don't know the history on this and was just trying to make the current dependencies correct. But good points... let me look into it more without blinders on.

Reply:
Looks like the GPU targets handle their deps on AMDGPU and NVPTX explicitly, so these are just for the CPU targets. I agree this could be better. Adding an explanation and TODO and proceeding as-is since it is an improvement over what it was (i.e. it is actually correct).